core/src/broadcast_stage/standard_broadcast_run.rs

```diff
@@ -62,6 +62,7 @@ impl StandardBroadcastRun {
                         None,
                         true,
                         true,
+                        max_ticks_in_slot & SHRED_TICK_REFERENCE_MASK,
```

Comment:
@pgarg66 Do you recall why we are using
```rust 
max_ticks_in_slot & SHRED_TICK_REFERENCE_MASK
```
for `reference_tick` here?
`max_ticks_in_slot` is set to `bank.ticks_per_slot()` below. So if this is equal to `DEFAULT_TICKS_PER_SLOT = 64` then the bit-wise and with `SHRED_TICK_REFERENCE_MASK` will set the `reference_tick` to zero, since the mask is equal to 63.

The other place in line 163 below is using 
```rust
(bank.tick_height() % bank.ticks_per_slot()) as u8
```
for `reference_tick`. So in this line for interrupted slot, it seems to me we probably want to set `reference_tick` to something like `bank.ticks_per_slot()` or  `bank.ticks_per_slot() - 1`.

Reply:
It's been a while since I looked at this part of the code. So my memory is a bit hazy.

IIRC, in `check_for_interrupted_slot()`, it's generating a shred to fill out the partial slot with the empty ticks. So the data shred reference should be the 0th tick in the partial slot. 

You are right that `max_ticks_in_slot & SHRED_TICK_REFERENCE_MASK` evaluates to 0. I am not certain why we did not just use 0 there instead of evaluating it from `max_ticks_in_slot`. One reason could be to add tests for different conditions. The only test I see is passing a 0. Maybe more were planned but never got added.